### ISSUE TITLE
Count support for text-object selection

### ISSUE THREAD

**Issue by andykitchen** [mawww/kakoune#3761]
### Feature 

Count argument support for text object selection e.g. via ']' and '<alt-a>'

### Usecase 

One would often like to select the next two or three sentences, paragraphs or other objects.
Currently you just need to press ']p' multiple times, the intuitive '3]p' currently selects just one paragraph,
instead of 3 which is a bit confusing and seems like an notable omission of a reasonably common operation.
It seems arbitrary that '3W' works as expected but '3]p' doesn't.

Hope this is a useful suggestion :)

AK

P.S. I'm currently on the Ubuntu 20.04 packaged version: 2019.07.01-1build1

P.P.S. Is it beneficial to make the count argument just mechanically repeat any command if there is no special handling?
It seems that any time I could repeatedly type a command, adding a count argument would be an efficient alternative
that does effectively the same thing (except perhaps for counting it as one operation from the point of view of undo).
One could certainly save the command as a macro and then repeat the macro, but that seems unnecessary.
There may be some extreme cases where a count argument truly is nonsense, but they are tricky to find, in a very quick
review of all the normal mode commands each seemed to either benefit from a repeat count or be idempotent (in which case the repeat count, would not hurt, but simply have no effect). Put another way, perhaps all non-idempotent commands
should be required to do something sensible with the command count.

**Comment by andykitchen** [comment]
After following a chain of PRs referencing each other it seems this may have been fixed earlier this year: https://github.com/mawww/kakoune/pull/3630

**Comment by Delapouite** [comment]
You're right Andy, the sentence and paragraph text-objects now support count thanks to the PR you linked.

They are a few remaining case not handled such as quotes, but it is tracked in another issue: https://github.com/mawww/kakoune/issues/795

So I think we can close this one.
